railties/lib/rails/commands/credentials/credentials_command.rb

```diff
@@ -20,6 +20,11 @@ def edit
         require_application_and_environment!
 
         ensure_editor_available(command: "bin/rails credentials:edit") || (return)
+
+        if Rails.application.credentials.key.nil?
+          return say missing_credentials_message if Rails.application.credentials.content_path.exist?
+        end
```

Comment:
This breaks the flow of the command. Before we had 4 lines of `ensure_*` methods but now there's something here at a more granular level of abstraction.

I don't think `missing_credentials_message` makes sense here either. It's `else` branch mentions using `credentials:edit`, which is what the user is already running. Confusing usage.

Aren't both checks just variants of `ensure_master_key_has_been_added`? If we haven't found a key we should add it but only if we're generating new credentials?

Reply:
@kaspth

Thanks for the comment! The way of abstraction sounds reasonable for me, but I'm not completely sure what you meant.

You prefer to prepare another message but `missing_credentials_message` and abstract the condition `Rails.application.credentials.key.nil? && !Rails.application.credentials.content_path.exist?` as follows:

```ruby
def edit
  .... 
  return say "some messages" unless master_key_should_be_added?
  ....
end

private
  def master_key_should_be_added?
    Rails.application.credentials.key.nil? && !Rails.application.credentials.content_path.exist?
  end
```

Reply:
I'm not convinced we need another message. What do we need to explain to the user here? That we aren't doing anything?

I was alluding to just doing this:

```ruby
def ensure_master_key_has_been_added
  if Rails.application.credentials.key.nil? && !Rails.application.credentials.content_path.exist?
    master_key_generator.add_master_key_file
    master_key_generator.ignore_master_key_file
  end
end
```